src/gap-skipper.js

```diff
@@ -144,6 +144,60 @@ export default class GapSkipper {
     this.tech_.setCurrentTime(nextRange.start(0) + Ranges.TIME_FUDGE_FACTOR);
   }
 
+  gapFromVideoUnderflow_(buffered, currentTime) {
+    // At least in Chrome, if there is a gap in the video buffer, the audio will continue
+    // playing for ~3 seconds after the video gap starts. This is done to account for
+    // video buffer underflow/underrun (note that this is not done when there is audio
+    // buffer underflow/underrun -- in that case the video will stop as soon as it
+    // encounters the gap, as audio stalls are more noticeable/jarring to a user than
+    // video stalls). The player's time will reflect the playthrough of audio, so the
+    // time will appear as if we are in a buffered region, even if we are stuck in a
+    // "gap."
+    //
+    // Example:
+    // video buffer:   0 => 10.1, 10.2 => 20
+    // audio buffer:   0 => 20
+    // overall buffer: 0 => 10.1, 10.2 => 20
+    // current time: 13
+    //
+    // Chrome's video froze at 10 seconds, where the video buffer encountered the gap,
+    // however, the audio continued playing until it reached ~3 seconds past the gap
+    // (13 seconds), at which point it stops as well. Since current time is past the
+    // gap, findNextRange will return no ranges. To check for this issue, we see if
+    // we can get a next range from 4 seconds back (4 seconds to account for variance).
+    let nextRangeFromPossibleGap = Ranges.findNextRange(buffered, currentTime - 4);
+
+    if (nextRangeFromPossibleGap.length === 0) {
```

Comment:
You're right. I tried to separate steps a little bit for clarity (and possibly for a bit of efficiency), but it may not even be much more efficient, and it ends up being clearer without this step. Removing.